### ISSUE TITLE
Network Unavailable Bug

### ISSUE THREAD

**Issue by CaptainKhan** [mentaljam/harbour-storeman#50]
Gets stuck on Network Unavailable message. Please allow MANUAL network refresh or reset on app opening.

**Comment by mentaljam** [comment]
Storeman uses NetworkManager from Connman to watch the network state. Did you change any Connman settings?

**Comment by wickedsp1d3r** [comment]
Could this be changed/overridden somehow?
I am using separate DNS server and I can't connect to network with StoreMan.

**Comment by mentaljam** [comment]
Does Storeman show you the "Network unavailable" message or it just cannot load any page?

**Comment by mentaljam** [comment]
@wickedsp1d3r, also can you please run `connmanctl state` and send me the output?

**Comment by wickedsp1d3r** [comment]
> Does Storeman show you the "Network unavailable" message or it just cannot load any page?

Yes, when I open the application or try to open a category. all I get is a message "Network unavailable". "Refresh" option is disabled.

> @wickedsp1d3r, also can you please run `connmanctl state` and send me the output?

```
State = ready
OfflineMode = False
SessionMode = False
InputRequestTimeout = 300000
```

**Comment by mentaljam** [comment]
The state is `ready` which means a connected device:

https://github.com/aldebaran/connman/blob/914cf2749d4fc24b54f4a0c44f690090d08af147/doc/service-api.txt#L122

But Storeman expects the 'online` state:

https://github.com/mentaljam/harbour-storeman/blob/cdbd1e623724ebd5c3cc50f9f4eade3e328a5a9d/qml/harbour-storeman.qml#L87

You need check the `EnableOnlineCheck` and `Ipv4StatusUrl` in your `/etc/connman/main.conf`:
- `EnableOnlineCheck` must be `true` or undefined
- `Ipv4StatusUrl` must point to an address your device can resolve and access

**Comment by speefak** [comment]
harbour-storeman 
+ harbour-storeman
[D] unknown:0 - Using Wayland-EGL
[W] unknown:0 - Theme dir "/usr/share/themes/jolla/meegotouch/z1.25/" does not exist
[C] unknown:0 - Could not read "/home/.zypp-cache/solv/@System/solv"
[W] unknown:0 - Network is unavailable!

Network on device works, browser too. Permission issue caused by firejail ? Do i need a firejail profile ? Where i can get detailed information about the connecetion or permission error ?

**Comment by speefak** [comment]
solved : updated SFOS to 4.1
